src/topology/builder.rs

```diff
@@ -246,10 +246,16 @@ impl<'a> Builder<'a> {
                 let mut rx = builder.add_source_output(output.clone());
 
                 let (mut fanout, control) = Fanout::new();
+                let source = Arc::new(OutputId {
+                    component: key.clone(),
+                    port: output.port.clone(),
```

Comment:
Is the output port actually used/useful? What is the envisioned use case?

Reply:
That's a good question. The primary use case is for the telemetry that the sinks will emit. Adding the source as a tag to these metrics will allow an event to be tracked from source to sink. 

A secondary benefit will be to give VRL access to the field, something that has been asked for a number of times.

As it stands at the minute, the port is probably not that useful since the only sources that specify a port at the moment use the port to distinguish between logs, traces or metric outputs. So the port can be determined through other means. I'm not sure whether this will always be the case though..

Reply:
I'm 50/50 on this one. If there's not a known use, I don't think we should include it from the basic YAGNI principle. However, since it's in an `Arc`, cloning it costs no more than cloning the base component name, so it's pretty much free.

Reply:
I will shortly be adding another field to the metadata to indicate the previous component in the pipeline. That field will need the port, since the output of the port determines the schema definition that the component outputs. Keeping it like this does make it consistent with that. 

But still I am also 50/50 on this.